python/ray/util/state/state_manager.py

```diff
@@ -293,10 +293,6 @@ async def get_all_task_info(
             else:
                 continue
 
-            # Remove the filter from the list so that we don't have to
-            # filter it again later.
-            filters.remove(filter)
-
```

Comment:
Is this intentional change? Why is this necessary?

Reply:
Yes, and this bug is hard to find. I discussed this behavior in the following context:

> Backend: Synced with @rickyyx, we removed the optimization that distributed some filters in the GCS layer to decrease the latency of filtering.

Previously, we had an optimization for filters where we distributed some filters in the GCS layer to decrease the latency of filtering. For example, if we have two filters, task_id, and state:

Previously:
`GCS -> stateAPI (applied 2 filter conditions) -> API router
`

After the optimization:
`GCS (1 filter applied) -> stateAPI (1 filter applied) -> API router
`

Applying a filter at each layer is faster than applying all filters at the stateAPI layer. However, this will make the API `get_all_task_info` not pure since it modifies the data after searching. Therefore, we removed the filter optimization here.

Of course, we could deep copy the filters here, but after syncing with @rickyyx, we agreed to remove the filter optimization here.